ev3dev2/console.py

```diff
@@ -0,0 +1,135 @@
+        is 14 columns and 5 rows.
+
+        Using the `inverse=True` parameter will display the `text` with more emphasis and contrast,
+        as the background of the text will be black, and the foreground is white. Using inverse
+        can help in certain situations, such as to indicate when a color sensor senses
+        black, or the gyro sensor is pointing to zero.
+
+        Use the `alignment` parameter to enable the function to align the `text` differently to the
+        column/row values passed-in. Use `L` for left-alignment (default), where the first character
+        in the `text` will show at the column/row position. Use `R` for right-alignment, where the
+        last character will show at the column/row position. Use `C` for center-alignment, where the
+        text string will centered at the column/row position (as close as possible using integer
+        division--odd-length text string will center better than even-length).
+
+        Parameters:
+
+        - `text` (string): Text to display
+        - `column` (int): LCD column position to start the text (1 = left column);
+          text will wrap when it reaches the right edge
+        - `row` (int): LCD row position to start the text (1 = top row)
+        - `reset_console` (bool): ``True`` to reset the EV3 LCD console before showing
+          the text; default is ``False``
+        - `inverse` (bool): ``True`` for white on black, otherwise black on white;
+          default is ``False``
+        - `alignment` (string): Align the `text` horizontally. Use `L` for left-alignment (default),
+          `R` for right-alignment, or `C` for center-alignment
+
+        """
+
+        if reset_console:
+            self.reset_console()
+
+        if alignment == "R":
+            column = column - len(text) + 1
+        elif alignment == "C":
+            column -= len(text) // 2
+
+        if inverse:
+            text = "\x1b[7m{}\x1b[m".format(text)
+
+        print("\x1b[{};{}H{}".format(row, column, text), end='')
+
+    def set_font(self, font, reset_console=True):
+        """
+        Set the EV3 LCD console font and optionally reset the EV3 LCD console
+        to clear it and turn off the cursor.
+        Parameter:
+
+        - `font` (string): Font name, as found in `/usr/share/consolefonts/`
+        - `reset_console` (bool): ``True`` to reset the EV3 LCD console
+          after the font change; default is ``True``
+
+        """
+        if font is not None and font != self._font:
+            self._font = font
+            os.system("setfont {}".format(font))
+
+        if reset_console:
+            self.reset_console()
+
+    def show_cursor(self, on=False):
+        """
+        Use ANSI codes to turn the EV3 LCD console cursor on or off
+
+        Parameter:
+
+        - `on` (bool): ``True`` to turn on the cursor; default is ``False``
+
+        """
+        print("\x1b[?25{}".format('h' if on else 'l'), end='')
+
+    def clear_to_eol(self):
```

Comment:
Is there a way to move the cursor so I could go back and clear a specific line?

Reply:
Good idea. Added column, row parameters to enable positioning before clearing... defaults to None to simply clear from current cursor position.